app/frontend/components/domains/users/invite-screen.tsx

```diff
@@ -0,0 +1,234 @@
+import { SharedSpinner } from "../../shared/base/shared-spinner"
+import { RouterLink } from "../../shared/navigation/router-link"
+
+interface IInviteScreenProps {}
+
+type TFormData = {
+  users: { firstName?: string; lastName?: string; email?: string; role: EUserRoles; jurisdictionId: string }[]
+}
+
+export const InviteScreen = observer(({}: IInviteScreenProps) => {
+  const { t } = useTranslation()
+  const { jurisdiction, error } = useJurisdiction()
+  const {
+    userStore: { invite },
+  } = useMst()
+
+  const defaultUserValues = {
+    role: EUserRoles.reviewManager,
+    jurisdictionId: jurisdiction?.id,
+  }
+
+  const formMethods = useForm<TFormData>({
+    mode: "onChange",
+    defaultValues: {
+      users: [defaultUserValues],
+    },
+  })
+
+  const { handleSubmit, formState, control } = formMethods
+
+  const { fields, append, remove } = useFieldArray({
+    control,
+    name: "users",
+  })
+
+  const { isSubmitting } = formState
+
+  const onSubmit = async (formData) => {
+    invite(formData)
+  }
+
+  const navigate = useNavigate()
+
+  if (!jurisdiction) {
+    return (
+      <Flex as="main" w="full" bg="greys.white">
+        <SharedSpinner />
+      </Flex>
+    )
+  }
+
+  return (
+    <Flex direction="column" w="full" bg="greys.white">
+      <Container maxW="container.lg" py={16} px={8}>
+        <Flex direction="column" gap={8}>
+          <Flex direction="column">
+            <Heading as="h1">{t("user.inviteTitle")}</Heading>
+            <Text>
+              {t("user.inviteInstructions")} <RouterLink to="#">{t("user.rolesAndPermissions")}</RouterLink>
+            </Text>
+          </Flex>
+          <Heading fontSize="2xl">{jurisdiction.name}</Heading>
+          <FormProvider {...formMethods}>
+            <form onSubmit={handleSubmit(onSubmit)}>
+              <Flex direction="column" gap={6}>
+                <Flex direction="column" gap={4}>
+                  {fields.map((field, index) => (
+                    <UserInput key={field.id} index={index} remove={remove} />
+                  ))}
+                  <Button
+                    type="button"
+                    variant="tertiary"
+                    onClick={() => append(defaultUserValues)}
+                    leftIcon={<FontAwesomeIcon style={{ height: 14, width: 14 }} icon={faPlus} />}
+                  >
+                    {t("user.addUser")}
+                  </Button>
+                </Flex>
+                <Flex gap={4}>
+                  <Button
+                    variant="primary"
+                    type="submit"
+                    isLoading={isSubmitting}
+                    loadingText={t("ui.loading")}
+                    rightIcon={<FontAwesomeIcon style={{ height: 14, width: 14 }} icon={faPaperPlane} />}
+                  >
+                    {t("user.sendInvites")}
+                  </Button>
+                  <Button variant="secondary" isLoading={isSubmitting} onClick={() => navigate(-1)}>
```

Comment:
What if a user pastes a link and directly navigates to this page? The `navigate(-1)` might not work. We should first check if the there is a page before, if not we should navigate to a default page